public/video-ui/src/slices/s3Upload.ts

```diff
@@ -0,0 +1,50 @@
+import { createSlice,PayloadAction } from '@reduxjs/toolkit';
+
+export interface S3UploadState {
+  id: string | null;
+  progress: number;
+  total: number;
+  status: 'idle' | 'uploading' | 'post-processing' | 'complete' | 'error';
+}
+
+type S3UploadAction = PayloadAction<S3UploadState>
+
+const initialState : S3UploadState =  { id: null, progress: 0, total: 0, status: 'idle' };
+
+const s3Upload = createSlice({
+  name: 's3Upload',
+  initialState,
+  reducers: {
+    s3UploadStarted:(state :S3UploadState, action)=> {
+      const total = action.payload.parts[action.payload.parts.length - 1].end;
+      Object.assign(state, {
+        id: action.payload.id,
+        total: total,
+        status: 'uploading'
+      });
+    },
+    s3UploadProgress: (state : S3UploadState, action: S3UploadAction)=>{
+      Object.assign(state, {
+        progress: action.payload.progress,
+        status: 'uploading'
+      });
+      },
+      s3UploadPostProcessing: _=> ({
```

Comment:
Instead of having the reducer methods return a whole `S3UploadState`, they can also modify the property of the current value, which would probably be a bit clearer in these reducers that just need to change the status eg:
```
    s3UploadPostProcessing: (state) => {
      state.status = 'post-processing';
    },
```

This is just another stylistic point, but if the methods had "verb" names like "setS3UploadStatusToPostProcessing", it would make it a bit clearer what dispatching them would do